### PR TITLE
Update dependency go modules for k8s v1.28.0

### PR DESCRIPTION
**Pull request by Sneha-at**
Ran kubernetes-csi/csi-release-tools go-get-kubernetes.sh -p 1.28.0
```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,9 +11,9 @@ require (
 	github.com/kubernetes-csi/csi-lib-utils v0.14.0
 	github.com/kubernetes-csi/csi-test/v4 v4.4.0
 	google.golang.org/grpc v1.57.0
-	k8s.io/api v0.28.0-rc.0
-	k8s.io/apimachinery v0.28.0-rc.0
-	k8s.io/client-go v0.28.0-rc.0
+	k8s.io/api v0.28.0
+	k8s.io/apimachinery v0.28.0
+	k8s.io/client-go v0.28.0
 	k8s.io/csi-translation-lib v0.28.0-rc.0
```
Comment (Sneha-at) [inline go.mod:17]: Tag `v0.28.0` is not yet created for [csi-transition-lib](https://pkg.go.dev/k8s.io/csi-translation-lib?tab=versions)

File: go.sum [hunk 2]
```diff
@@ -314,12 +314,12 @@ gopkg.in/yaml.v3 v3.0.1 h1:fxVm/GzAzEWqLHuvctI91KS9hhNmmWOoWu0XTYJS7CA=
...
-k8s.io/client-go v0.28.0-rc.0 h1:fL9kSASMgDyDD9uVMevQR0aqBY7Mczb/GzsqlUNwq4Q=
-k8s.io/client-go v0.28.0-rc.0/go.mod h1:KZq8b8mjSNmYMofEo2M7XEehqaonyBZFw5fdsups46Y=
+k8s.io/api v0.28.0 h1:3j3VPWmN9tTDI68NETBWlDiA9qOiGJ7sdKeufehBYsM=
+k8s.io/api v0.28.0/go.mod h1:0l8NZJzB0i/etuWnIXcwfIv+xnDOhL3lLW919AWYDuY=
+k8s.io/apimachinery v0.28.0 h1:ScHS2AG16UlYWk63r46oU3D5y54T53cVI5mMJwwqFNA=
+k8s.io/apimachinery v0.28.0/go.mod h1:X0xh/chESs2hP9koe+SdIAcXWcQ+RM5hy0ZynB+yEvw=
+k8s.io/client-go v0.28.0 h1:ebcPRDZsCjpj62+cMk1eGNX1QkMdRmQ6lmz5BLoFWeM=
+k8s.io/client-go v0.28.0/go.mod h1:0Asy9Xt3U98RypWJmU1ZrRAGKhP6NqDPmptlAzK2kMc=
 k8s.io/component-base v0.26.0 h1:0IkChOCohtDHttmKuz+EP3j3+qKmV55rM9gIFTXA7Vs=
 k8s.io/component-base v0.26.0/go.mod h1:lqHwlfV1/haa14F/Z5Zizk5QmzaVf23nQzCwVOQpfC8=
```
Comment (Sneha-at) [inline go.sum:324]: This is dependent on `csi-lib-utils` (v0.14.0) which still uses `k8s.io/component-base v0.26.0`. The latest version available for `csi-lib-utils` is v0.15.0-rc0

File: go.mod (hunk 1 above)
Comment (jsafrane) [reply go.mod:17]: @Sneha-at, can you please fix csi-translation-lib version to v0.28.0? The PR is good otherwise.

Comment (Sneha-at) [reply go.mod:17]: Sure, I thought we are waiting for the fix to add the version on `pkg.go.dev/k8s.io/csi-translation-lib`. I will update the package version here.

File: go.sum (hunk 2 above)
Comment (Sneha-at) [reply go.sum:324]: @jsafrane What do you recommend for this?

**Pull request merged**